src/compas/tolerance.py

```diff
@@ -151,22 +158,34 @@ def __init__(
         name=None,
     ):
         super(Tolerance, self).__init__(name=name)
-        self._unit = None
-        self._absolute = None
-        self._relative = None
-        self._angular = None
-        self._approximation = None
-        self._precision = None
-        self._lineardeflection = None
-        self._angulardeflection = None
-        self.unit = unit
-        self.absolute = absolute
-        self.relative = relative
-        self.angular = angular
-        self.approximation = approximation
-        self.precision = precision
-        self.lineardeflection = lineardflection
-        self.angulardeflection = angulardflection
+        if not self._is_inited:
+            self._unit = None
+            self._absolute = None
+            self._relative = None
+            self._angular = None
+            self._approximation = None
+            self._precision = None
+            self._lineardeflection = None
+            self._angulardeflection = None
+
+        self._is_inited = True
+
+        if unit is not None:
```

Comment:
not important.... but i guess unit will never be None because it has a default "M", no?